packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -315,34 +311,36 @@ export function FileUploader({
     [acceptedFileTypes, fileStatuses, setFileStatuses]
   );
 
+  const updateEditStatus = useCallback(
```

Comment:
Think you can probably move the call to `setFileStatuses` required by `onCancelEdit` and `onStartEdit` directly in to this function and pass the logic here in to `setFileStatuses` to make use of the previous value instead of using `fileStatuses` directly in this component. 

The benefit of using the previous value is that we no longer need to include `fileStatuses` in the dependency array, which prevents `updateEditStatus`, `onCancelEdit` and `onStartEdit` from reevaluating on each update to `fileStatuses`.

Last, consider replacing `isCancelEdit` with a param of `fileState`, which you can just pass directly on line 320. That will make this function easier to extend from in the future and allows this function to have a more generic name of `updateFileState`

Reply:
👍

Reply:
[885920e](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/885920e203a8c6326d6086725c05a11f0b2e602f)